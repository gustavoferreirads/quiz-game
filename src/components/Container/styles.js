import { StyleSheet } from 'react-native';
import { metrics } from '../../styles';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: metrics.size,
  },
});
