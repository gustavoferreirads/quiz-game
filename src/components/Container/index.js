import React from 'react';
import { View } from 'react-native';
import * as PropTypes from 'prop-types';

import styles from './styles';

const Container = ({ children, style }) => (
  <View style={[styles.container, style]}>{children}</View>
);
Container.defaultProps = {
  style: {},
};
Container.propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.array]).isRequired,
  style: PropTypes.object,
};

export default Container;
