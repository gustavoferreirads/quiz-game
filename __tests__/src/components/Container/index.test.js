import React from 'react';
import { Text } from 'react-native';
import { shallow } from 'enzyme/build';
import Container from '../../../../src/components/Container';

it('renders a Container ', () => {
  expect(
    shallow(
      <Container style={{ backgroundColor: 'red' }}>
        <Text>Label</Text>
      </Container>,
    ),
  ).toMatchSnapshot();
});
