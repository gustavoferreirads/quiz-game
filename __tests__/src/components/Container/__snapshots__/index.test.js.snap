// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Container  1`] = `
<Component
  style={
    Array [
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 10,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
>
  <Component>
    Label
  </Component>
</Component>
`;
